api/src/test/java/org/apache/iceberg/util/RandomUtil.java

```diff
@@ -31,6 +31,12 @@
   private RandomUtil() {
   }
 
+  private static final Random NEGATIVE_VALUES = new Random();
```

Comment:
All tests need to use the `Random` that is passed in so that the values that are generated are deterministic and repeatable. That's what allows us to generated an iterator instead of a list in some cases, and makes it so if we do have a problem, we can repeat the test and debug.

Reply:
For some reason, TestLocalScan#testRandomData() fails if I use the random passed in to do
```
boolean negate = random.nextInt(2) == 1
```

Reply:
Looks like this caused a test failure in TestSparkParquetWriter#testCorrectness also. Pushed a commit to get rid of the new Random.